File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        int[] answer = new int[2];
        int count = 1;
        
        for (int i = 1; i <= n; i++) {
            if (n % i == 0 && m % i == 0) {
                answer[0] = i;                
            }
        }
        
        while (true) {            
            if ((n * count) % m == 0) {
                answer[1] = n * count;
                break;
            }
            count++;            
        }
        
        return answer;
    }
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
function solution(n, m) {
    var answer = [];   
    let count = 1;
    
    if (n < m) {
        for (let i = 1; i <= n; i++) {
            if (n % i === 0 && m % i === 0) {
                answer[0] = i;
            }
        }
    } else {
        for (let i = 1; i <= m; i++) {
            if (n % i === 0 && m % i === 0) {
                answer[0] = i;
            }
        }
    }
    
    while (isNaN(answer[1])) {
        if ((n * count) % m === 0) {
            answer[1] = n * count;
        }
        count++;
    }    
    
    return answer;
}